var/spack/repos/builtin/packages/neurodamus-model/package.py

```diff
@@ -82,6 +82,10 @@ def setup_common_mods(self, spec, prefix):
         if spec.variants['common_mods'].value != 'default':
             shutil.move('common', '_common_orig')
             force_symlink(spec.variants['common_mods'].value, 'common')
+        else:
+            # Newer models require fecthing common when working standalone
+            fetch_common = which("./fetch_common.bash")
```

Comment:
What does this fetch? Will it do a bunch of `git` stuff, or just copy files around?

Reply:
Basically a git clone since common is not found one level up.

Reply:
Can we add that as a resource, then?

Reply:
For some reasons I'd prefer not to.
this package is inherited by several specific ones it's hard to know when if we should fetch the resource. 
The script also checks if the clone is actually needed since there's already some links, which, in case they's not sufficient, will delete. 
This way we also test the script which is there anyway for users.

Reply:
We can, but then previous versions will be compiled differently too... If that's acceptable, ok w me

Reply:
I would also add as resource than running shell script. Could you please explain a bit following?

> We can, but then previous versions will be compiled differently too

If this is after re-structuring repos, I assume it's not cleaner to maintain the login in the Spack recipe?

If this is more complex, I think we should have quick sync over zoom. (Just bit more curious/serious about the reproducibility aspects for main simulation toolchain i.e. neurodamus)

Reply:
> We can, but then previous versions will be compiled differently too

Models don't use submodule anymore.
if we re-build say neurodamus-neocortex@3.1.0 it may be built differently than the existing 3.1.0, which was using submodules. This can be very hard to debug if results change whatsoever.

>  I assume it's not cleaner to maintain the login in the Spack recipe?

Login? We have no probs with authorization, works one way or another.

Reply:
>  Login? We have no probs with authorization, works one way or another.

Sorry, I meant `logic` of shell script `./fetch_common.bash`. 

> Models don't use submodule anymore.

How submodule works? If I clone older version/tag is checked out, that won't contain the submodule?

Reply:
Yes and the resource would overwrite the submodule. This is handled by the simple script, if exists. I didn't want to introduce more logic in the recipe. But maybe there are missing pieces of info here. Let's discuss tomorrow.

Reply:
The resource can be gated with `when=…`?

Reply:
Thought so too initially, but `when=...` works normally with version, here we don't know the version, that comes only in the actual package, not in the superclass

Reply:
I guess you'r afraid of scripts to avoid a situation where magically CI does something different, or fails, which I understand. However this script really does only a check if it can access `common` and if not then clones:
https://bbpgitlab.epfl.ch/hpc/sim/models/neocortex/-/blob/main/fetch_common.bash
In case we can copy paste the code to the spack recipe.

Reply:
So it just unconditionally fetches the latest HEAD? How are we supposed to create reproducible builds then?